server/src/main/java/org/elasticsearch/index/codec/vectors/reflect/OffHeapReflectionUtils.java

```diff
@@ -0,0 +1,147 @@
+    static final String HNSW_VECTOR_INDEX_EXTENSION = "vex";
+
+    private static final MethodHandle GET_FIELD_ENTRY_HNDL_SQ;
+    private static final MethodHandle GET_VECTOR_DATA_LENGTH_HANDLE_SQ;
+    private static final VarHandle RAW_VECTORS_READER_HNDL_SQ;
+    private static final MethodHandle GET_FIELD_ENTRY_HANDLE_L99FLT;
+    private static final MethodHandle VECTOR_DATA_LENGTH_HANDLE_L99FLT;
+    private static final MethodHandle GET_FIELD_ENTRY_HANDLE_L99HNSW;
+    private static final MethodHandle GET_VECTOR_INDEX_LENGTH_HANDLE_L99HNSW;
+    private static final VarHandle FLAT_VECTORS_READER_HNDL_L99HNSW;
+
+    static final Class<?> L99_SQ_VR_CLS = Lucene99ScalarQuantizedVectorsReader.class;
+    static final Class<?> L99_FLT_VR_CLS = Lucene99FlatVectorsReader.class;
+    static final Class<?> L99_HNSW_VR_CLS = Lucene99HnswVectorsReader.class;
+
+    static {
+        try {
+            // Lucene99ScalarQuantizedVectorsReader
+            var cls = Class.forName("org.apache.lucene.codecs.lucene99.Lucene99ScalarQuantizedVectorsReader$FieldEntry");
+            var lookup = privilegedPrivateLookupIn(L99_SQ_VR_CLS, MethodHandles.lookup());
+            var mt = methodType(cls, String.class);
+            GET_FIELD_ENTRY_HNDL_SQ = lookup.findVirtual(L99_SQ_VR_CLS, "getFieldEntry", mt);
+            GET_VECTOR_DATA_LENGTH_HANDLE_SQ = lookup.findVirtual(cls, "vectorDataLength", methodType(long.class));
+            RAW_VECTORS_READER_HNDL_SQ = lookup.findVarHandle(L99_SQ_VR_CLS, "rawVectorsReader", FlatVectorsReader.class);
+            // Lucene99FlatVectorsReader
+            cls = Class.forName("org.apache.lucene.codecs.lucene99.Lucene99FlatVectorsReader$FieldEntry");
+            lookup = privilegedPrivateLookupIn(L99_FLT_VR_CLS, MethodHandles.lookup());
+            mt = methodType(cls, String.class, VectorEncoding.class);
+            GET_FIELD_ENTRY_HANDLE_L99FLT = lookup.findVirtual(L99_FLT_VR_CLS, "getFieldEntry", mt);
+            VECTOR_DATA_LENGTH_HANDLE_L99FLT = lookup.findVirtual(cls, "vectorDataLength", methodType(long.class));
+            // Lucene99HnswVectorsReader
+            cls = Class.forName("org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsReader$FieldEntry");
+            lookup = privilegedPrivateLookupIn(L99_HNSW_VR_CLS, MethodHandles.lookup());
+            mt = methodType(cls, String.class, VectorEncoding.class);
+            GET_FIELD_ENTRY_HANDLE_L99HNSW = lookup.findVirtual(L99_HNSW_VR_CLS, "getFieldEntry", mt);
+            GET_VECTOR_INDEX_LENGTH_HANDLE_L99HNSW = lookup.findVirtual(cls, "vectorIndexLength", methodType(long.class));
+            lookup = privilegedPrivateLookupIn(L99_HNSW_VR_CLS, MethodHandles.lookup());
+            FLAT_VECTORS_READER_HNDL_L99HNSW = lookup.findVarHandle(L99_HNSW_VR_CLS, "flatVectorsReader", FlatVectorsReader.class);
+        } catch (ReflectiveOperationException e) {
+            throw new AssertionError(e);
```

Comment:
It looks like this class won't be initialized until the first call to get bytes. Could we more aggressively load this class so that we find any misformulated method handle lookups during startup?

Reply:
yeah, we can eagerly initialise it. Done in [0a198b5](https://github.com/elastic/elasticsearch/pull/126704/commits/0a198b588b5b8bcccbe37094d37f34db5f0e4bb7)